R/extract_data.R

```diff
@@ -0,0 +1,245 @@
+    dplyr::rename(plot_context_id = link_vals)
+  
+  database$treatment_context_id <- database$contexts_tmp$treatment_context_id %>%
+    dplyr::rename(treatment_context_id = link_vals)
+
+  
+  tables_to_cut <- c("locations", "entity_context_id", "method_context_id", "temporal_context_id", 
+                            "plot_context_id", "treatment_context_id", 
+                            "methods", "taxa", "taxonomic_updates", "contributors")
+  
+  tables_complete_path <- c("database$locations", "database$entity_context_id", 
+                            "database$method_context_id", "database$temporal_context_id", 
+                            "database$plot_context_id", "database$treatment_context_id", 
+                            "database$methods", "database$taxa", "database$taxonomic_updates", "database$contributors")
+  
+  # Empty list
+  ret <- list(
+    traits = dplyr::tibble(), 
+    locations = dplyr::tibble(),
+    contexts = dplyr::tibble(),
+    entity_context_id = dplyr::tibble(),
+    method_context_id = dplyr::tibble(),
+    temporal_context_id = dplyr::tibble(),
+    plot_context_id = dplyr::tibble(),
+    treatment_context_id = dplyr::tibble(),
+    methods = dplyr::tibble(),
+    excluded_data = dplyr::tibble(),
+    taxonomic_updates = dplyr::tibble(),
+    taxa = dplyr::tibble(),
+    contributors = dplyr::tibble()
+  )
+  
+  ret_tmp <- list()
+
+  # Cookie cutters
+  locations_cc <- c("dataset_id", "location_id")
+  entity_contexts_cc <- c("dataset_id", "entity_context_id")
+  temporal_contexts_cc <- c("dataset_id", "temporal_context_id")
+  method_contexts_cc <- c("dataset_id", "method_context_id")
+  plot_contexts_cc <- c("dataset_id", "plot_context_id")
+  treatment_contexts_cc <- c("dataset_id", "treatment_context_id")
+  methods_cc <- c("dataset_id", "trait_name", "method_id")
+  taxonomic_updates_cc <- c("dataset_id", "taxon_name", "original_name")
+  taxa_cc <- c("taxon_name")
+  contributors_cc <- c("dataset_id")
+  
+  cookie_cutters <- c("locations_cc", "entity_contexts_cc", "method_contexts_cc", "temporal_contexts_cc", "plot_contexts_cc", "treatment_contexts_cc", 
+                      "methods_cc", "taxa_cc", "taxonomic_updates_cc", "contributors_cc")
+  
+  # Create a table of various look-ups used below
+  tables <- dplyr::tibble(
+    cookie_cutters = cookie_cutters,
+    tables_to_cut = tables_to_cut,
+    tables_complete_path = tables_complete_path
+  )
+  
+  # If the context table is queried need to convert the word "contexts" into a vector that indicates which of the 5 context categories have matches
+  if (table == "contexts") {
+
+    table_tmp <- database$contexts %>% 
+      dplyr::filter(stringr::str_detect(database$contexts[[col]], col_value)) %>%
+      dplyr::distinct(link_id)
+    
+    table <- as.vector(table_tmp[[1]])
+
+    if (length(table) == 0) {
+      table <-  as.vector("treatment_context_id")
+    }
+  
+  }
+  
+  for (i in seq_along(1:length(table))) {
```

Comment:
Then here

`  for (i in seq_along(1:length(table))) {` becomes `  for (tab in seq_along(cookie_cutters)) {`

and below replace `table[[i]]` with `tab`

Reply:
No, this doesn't work, this is for `length(table)` (the various context properties found), not `length(tables)` (the table of info/cookie cutters names)

Reply:
Oh I see, sorry. 

if you do

```
for (tab in seq_along(table))
```
then you can replace `table[[I]]` with `tab`